### PR TITLE
Change Buffer::SetSubData() start and offset to chars

### PR DESCRIPTION
**Pull request by SenorBlanco**
Change the uint32_t offsets in Buffer::SetSubData() to be uint8_ts, rather than uint32_ts.
This makes it practical to do subloads of 16-bit index buffers, for example.

### CODE REVIEW

File: examples/ComputeBoids.cpp [hunk 1]
```diff
@@ -88,7 +88,7 @@ void initBuffers() {
             .GetResult();
 
         particleBuffers[i].SetSubData(0,
-            sizeof(Particle) * kNumParticles / sizeof(uint32_t),
+            sizeof(Particle) * kNumParticles,
             reinterpret_cast<uint32_t*>(initialParticles.data()));
```
Comment (Kangz) [inline examples/ComputeBoids.cpp]: Can we have the `const uint32_t*` changed to a `const uint8_t*` to be consistent with the size argument?

File: src/tests/end2end/BufferTests.cpp [hunk 2]
```diff
@@ -209,7 +209,7 @@ TEST_P(BufferSetSubDataTests, SmallDataAtOffset) {
 
     constexpr uint32_t kOffset = 2000;
     uint32_t value = 298371;
-    buffer.SetSubData(kOffset / 4, 1, &value);
+    buffer.SetSubData(kOffset, sizeof(value), &value);
```
Comment (Kangz) [inline src/tests/end2end/BufferTests.cpp]: Can we also have 1 byte setsubdata tests and at an offset that's not aligned?

File: examples/ComputeBoids.cpp (hunk 1 above)
Comment (Kangz) [reply examples/ComputeBoids.cpp]: This will also make the wire work correctly, right now it thinks size is in number of uint32_t

Comment (SenorBlanco) [reply examples/ComputeBoids.cpp]: Done.

File: src/tests/end2end/BufferTests.cpp (hunk 2 above)
Comment (SenorBlanco) [reply src/tests/end2end/BufferTests.cpp]: Done.

**Pull request merged**